### PR TITLE
Big pull req

### PR DESCRIPTION
**Pull request by emolinac**
List of changes:
- Kinematical variables' name convention update
- hipo2root : addition of optional argument "-s" to specify the use of simulation
- make_ntuples : addition of optional argument "-s" to specify the use of simulation, addition of hadronic variables in TNtuple

### CODE REVIEW

File: src/particle.c [hunk 1]
```diff
@@ -17,6 +17,10 @@ particle particle_init(int pid, int charge, double beta, int status, int sector,
     // Inherent vars.
     p.is_valid = true;
     p.is_trigger_electron = (pid == 11 && status < 0);
+    p.is_hadron = false;
+    if(!p.is_trigger_electron){
```
Comment (bleaktwig) [inline src/particle.c:21]: This if statement is serves no purpose --- the for loop below filters out electrons anyway!

File: lib/constants.h [hunk 2]
```diff
@@ -160,18 +164,40 @@ extern const char * S_VAR_LIST[VAR_LIST_SIZE];
...
+#define A_PT2      31
+#define S_PL2     "Pl2"         // GeV^2.
+#define R_PL2     "pl2"
+#define A_PL2      32
+#define S_PHIPQ   "#phi_{PQ}"       // Rad.
+#define R_PHIPQ   "phipq"
+#define A_PHIPQ    33
+#define S_THETAPQ "#theta_{PQ}"     // Rad.
+#define R_THETAPQ "thetapq"
+#define A_THETAPQ  34
```
Comment (bleaktwig) [inline lib/constants.h:199]: The addresses here are gonna collide with recent changes in the master branch. Please pull from master and update accordingly.

File: src/constants.c [hunk 3]
```diff
@@ -78,3 +81,8 @@ const double STD_RX[STDPLT_LIST_SIZE][2][2] = {
 const long STD_BX[STDPLT_LIST_SIZE][2] = {
         {200,200}, {200,100}, {200,200}, { 22, -1}, { 22, -1}, { 20, -1}, {200, -1}, {200,200}
 };
+
+// Pid arrays
+const int HPID_ARRAY[NHADRONS] = {
+        211, -211, 2212, 2112, 321, -321, 221, 223, 111, 311
```
Comment (bleaktwig) [inline src/constants.c:87]: Recent versions of the master branch have a map of `{int PID, double mass}` called `MASS`. You should iterate through the key list in that array instead of creating a new one.

File: src/hipo2root.c [hunk 4]
```diff
@@ -12,20 +17,31 @@
...
-    else if (run_no /   1000 == 0) sprintf(out_filename, "../root_io/000%d.root", run_no);
-    else if (run_no /  10000 == 0) sprintf(out_filename, "../root_io/00%d.root", run_no);
-    else if (run_no / 100000 == 0) sprintf(out_filename, "../root_io/0%d.root", run_no);
-    else                           sprintf(out_filename, "../root_io/%d.root", run_no);
+    int  run_no    = -1;
+    bool use_simul = false;
+    
+    if (hipo2root_handle_args_err(hipo2root_handle_args(argc, argv, &in_filename, &run_no, &use_simul),
+                                      &in_filename))
+            return 1;
```
Comment (bleaktwig) [inline src/hipo2root.c:28]: This is just me being nitpicky, but why did you change the indentation here? Please follow the program's convention of 4 spaces, even if only to maintain my sanity.

File: src/hipo2root.c [hunk 5]
```diff
@@ -12,20 +17,31 @@
...
+        else if (run_no /    100 == 0) sprintf(out_filename, "../root_io/0000%d.root", run_no);
+        else if (run_no /   1000 == 0) sprintf(out_filename, "../root_io/000%d.root", run_no);
+        else if (run_no /  10000 == 0) sprintf(out_filename, "../root_io/00%d.root", run_no);
+        else if (run_no / 100000 == 0) sprintf(out_filename, "../root_io/0%d.root", run_no);
+        else                           sprintf(out_filename, "../root_io/%d.root", run_no);
+    } else{
+        // Simul process
+        // Creates file in current folder
+        out_filename = (char *) malloc(5 + strlen(argv[argc - 1]) + 1);        
+        sprintf(out_filename, "%s.root", argv[argc - 1]);
```
Comment (bleaktwig) [inline src/hipo2root.c:43]: This is terrible practice. The point of using `getopt` in `io_handler` is to avoid the unsafe handling of arguments like this. Nothing assures you that the filename will be the last argument, only that it will be the positional argument!

File: src/io_handler.c [hunk 6]
```diff
@@ -41,13 +49,30 @@ int extractsf_handle_args(int argc, char ** argv, bool * use_fmt, int * nevents,
...
+    while ((opt = getopt(argc, argv, "s")) != -1) {
+        switch (opt) {
+            case 's': * use_simul = true;         break;
+            default:  return 1; // Bad usage of optional arguments.
+        }
+    }
+
+    // Handle positional arguments
     if (argc < 2) return 1;
     if (argc > 3) return 2;
```
Comment (bleaktwig) [inline src/io_handler.c:64]: I'm not 100% on this, but shouldn't this condition change due to the new optional argument? Does this run ok?

File: src/io_handler.c [hunk 7]
```diff
@@ -67,13 +92,22 @@ int handle_root_filename(char * input_file, int * run_no, double * beam_energy)
...
 }
 
+int handle_root_filename_simul(char * input_file, double * beam_energy) {
+    int chk = check_root_filename(input_file);
+    if (chk) return chk;
+
+    // Beam energy of simulation is fixed here!
+    *beam_energy = 11.;
+    return 0;
+}
```
Comment (bleaktwig) [inline src/io_handler.c:102]: Creating a new method with mostly the same code as the old one is not good practice. Maybe you can better implement this with smart argument handling and overloading? It's not super necessary, but I'd rather keep the code as clean as possible to avoid future sneaky bugs.

File: src/io_handler.c [hunk 8]
```diff
@@ -41,13 +49,30 @@ int extractsf_handle_args(int argc, char ** argv, bool * use_fmt, int * nevents,
...
+    }
+
+    // Handle positional arguments
     if (argc < 2) return 1;
     if (argc > 3) return 2;
 
-    * input_file = (char *) malloc(strlen(argv[1]) + 1);
-    strcpy(* input_file, argv[1]);
-    return handle_hipo_filename(* input_file, run_no);
+    * input_file = (char *) malloc(strlen(argv[argc - 1]) + 1);
```
Comment (bleaktwig) [inline src/io_handler.c:66]: This is scary. What if I call the program using: `hipo2root FILENAME -s`? Wouldn't that break the program?

Edit. It only made sense before because the filename is the only argument the program receives.

File: src/make_ntuples.c [hunk 9]
```diff
@@ -31,6 +32,7 @@ int run(char * in_filename, bool use_fmt, bool debug, int nevn, int run_no, doub
 
     // Create TTree and TNTuples.
     TTree   * t_in  = f_in->Get<TTree>("Tree");
+    if(t_in ==NULL) return 1;
```
Comment (bleaktwig) [inline src/make_ntuples.c:35]: Nice, thanks.

File: src/make_ntuples.c [hunk 10]
```diff
@@ -81,6 +83,17 @@ int run(char * in_filename, bool use_fmt, bool debug, int nevn, int run_no, doub
         }
 
         // Process DIS event.
+        // Checking existence of trigger electron in event
+        particle p_el = particle_init();
+        for (UInt_t pos = 0; pos < rtrk.index->size(); ++pos) {
+            p_el = use_fmt ? particle_init(&rpart, &rtrk, &ftrk, pos)
+                           : particle_init(&rpart, &rtrk, pos);
+            if(p_el.is_trigger_electron) break;
```
Comment (bleaktwig) [inline src/make_ntuples.c:91]: Initializing a whole particle just to see if it's a trigger electron is a bit much. Maybe perform this check before initializing and **then** initialize `p_el`.

File: src/make_ntuples.c [hunk 11]
```diff
@@ -81,6 +83,17 @@ int run(char * in_filename, bool use_fmt, bool debug, int nevn, int run_no, doub
...
+        }
+        // Guarding statement
+        if(!p_el.is_trigger_electron) continue;
```
Comment (bleaktwig) [inline src/make_ntuples.c:94]: I think we already discussed this, but it'd be better if you didn't drop a whole event if you can't find a trigger electron.

File: src/make_ntuples.c [hunk 12]
```diff
@@ -128,20 +141,19 @@ int run(char * in_filename, bool use_fmt, bool debug, int nevn, int run_no, doub
...
+                (Float_t) p.pz, (Float_t) P(p), (Float_t) theta_lab(p),
+                (Float_t) phi_lab(p), (Float_t) p.beta,
                 (Float_t) chi2, (Float_t) ndf,
                 (Float_t) pcal_E, (Float_t) ecin_E, (Float_t) ecou_E, (Float_t) tot_E,
                 (Float_t) htcc_nphe, (Float_t) ltcc_nphe,
                 (Float_t) tof,
-                (Float_t) Q2(p, beam_E), (Float_t) nu(p, beam_E), (Float_t) Xb(p, beam_E),
-                        (Float_t) W2(p, beam_E)
+                (Float_t) Q2(p, beam_E), (Float_t) nu(p, beam_E), (Float_t) Xb(p, beam_E), (Float_t) W2(p, beam_E),
+                (Float_t) zh(p, p_el, beam_E),(Float_t) Pt2(p, p_el, beam_E),(Float_t) Pl2(p, p_el, beam_E),(Float_t) phi_pq(p, p_el, beam_E),(Float_t) theta_pq(p, p_el, beam_E)
```
Comment (bleaktwig) [inline src/make_ntuples.c:156]: Max line length is 100 characters.

Comment (bleaktwig) [reply src/make_ntuples.c:156]: Maybe I should write a styleguide convention and forbid code that doesn't follow it to be merged? That'd be tortuously fun...

**Review by bleaktwig — CHANGES_REQUESTED** [review]
Good luck

File: src/particle.c (hunk 1 above)
Comment (emolinac) [reply src/particle.c:21]: Yeah, I think this "if" would make sense if the process inside its scope was computationally expensive.

File: src/constants.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (emolinac) [reply src/constants.c]: Removed hadronic array. Currently using the map.

File: lib/constants.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (emolinac) [reply lib/constants.h]: Done.

File: src/hipo2root.c (hunk 4 above)
Comment (emolinac) [reply src/hipo2root.c]: My IDE went fuckin' nuts. Solved it.

File: src/make_ntuples.c (hunk 9 above)
Comment (emolinac) [reply src/make_ntuples.c]: XoXo

File: src/make_ntuples.c (hunk 10 above)
Comment (emolinac) [reply src/make_ntuples.c]: The method used now is more "economic":
1.- Inside event
2.- Check the PID and status given by reconstruction
3.- If a trigger electron is found, then save the pindex and the pos
    a .- I kept the trigger electron filter (see next issue)
4.- Obtain variables of trigger electron given by pindex and pos and write to both trees
5.- Obtain the rest of particles

File: src/make_ntuples.c (hunk 11 above)
Comment (emolinac) [reply src/make_ntuples.c]: Lets fight to the death to solve this one. Shall we?

File: src/err_handler.c [hunk 13]
```diff
@@ -1,10 +1,11 @@
 #include "../lib/err_handler.h"
 
 int make_ntuples_usage() {
-    fprintf(stderr, "Usage: make_ntuples [-fd] [-n NEVENTS] file\n");
+    fprintf(stderr, "Usage: make_ntuples [-fd] [-n NEVENTS] [-s] file\n");
```
[the remark this answers is not in the document]
Comment (bleaktwig) [reply src/err_handler.c]: It doesn't look done? But ok, it isn't necessary to approve the pull request.

**Comment by bleaktwig** [comment]
I'm gonna perform some tests after lunch and if everything looks ok I'll accept the pull request. Please make sure that you resolved all the requested changes, I see some unattended comments.

File: src/make_ntuples.c [hunk 14]
```diff
@@ -62,14 +62,26 @@ double get_tof(REC_Scintillator rsci, REC_Calorimeter  rcal, int pindex) {
...
-    // Extract sampling fraction parameters.
-    double sf_params[NSECTORS][SF_NPARAMS][2];
-    if (get_sf_params(Form("../data/sf_params_%06d.root", run_no), sf_params)) return 8;
+int run(char * in_filename, bool use_simul,bool debug, int nevn, int run_no, double beam_E) {
+    double sf_params[NSECTORS][SF_NPARAMS][2] = {{{0.246,0.0140},{0.985,1.},{0.083,0.},{-0.031,0.}},
+                                                 {{0.255,0.0150},{0.985,1.},{0.043,0.},{-0.024,0.}},
+                                                 {{0.245,0.0130},{0.985,1.},{0.067,0.},{-0.025,0.}},
+                                                 {{0.245,0.0140},{0.985,1.},{0.048,0.},{-0.016,0.}},
+                                                 {{0.252,0.0130},{0.985,1.},{0.007,0.},{-0.007,0.}},
+                                                 {{0.242,0.0140},{0.985,1.},{0.095,0.},{-0.038,0.}}};
```
Comment (bleaktwig) [inline src/make_ntuples.c:71]: What is this? Please add these numbers to a pre-prepared file in the data directory instead of hardcoding them.

File: src/extract_sf.c [hunk 15]
```diff
@@ -222,8 +223,10 @@ int run(char *in_filename, bool use_fmt, int nevn, int run_no) {
 
                 // Only add points within PLIMITSARR borders and with an acceptable chi2.
                 if ((mean - 2*sigma > PLIMITSARR[ci][0] && mean + 2*sigma < PLIMITSARR[ci][1]) &&
-                        (sf_gaus->GetChisquare() / sf_gaus->GetNDF() < SF_CHI2CONFORMITY))
-                    sf_dotgraph[ci][si]->AddPoint(p + SF_PSTEP/2, mean);
+                    (sf_gaus->GetChisquare() / sf_gaus->GetNDF() < SF_CHI2CONFORMITY)){
+                        // Older root compatibility fix
+                        sf_dotgraph[ci][si]->SetPoint(point_index, p + SF_PSTEP/2, mean); point_index++;
+                    }
                 }
```
Comment (bleaktwig) [inline src/extract_sf.c:230]: Indentation and the positioning of brackets here is wrong, please check.

File: src/extract_sf.c [hunk 16]
```diff
@@ -240,7 +243,14 @@ int run(char *in_filename, bool use_fmt, int nevn, int run_no) {
     }
 
     // Create output file.
-    TFile *f_out = TFile::Open(Form("../root_io/sf_study_%06d.root", run_no), "RECREATE");
+    char*  out_filename = (char *) malloc(128 * sizeof(char));
+    if(!use_simul){
+        sprintf(out_filename, "../root_io/sf_study_banks_%06d.root", run_no);
```
Comment (bleaktwig) [inline src/extract_sf.c:248]: This isn't working for data, there seems to be a mistake beforehand when saving the run_no variable.

Comment (bleaktwig) [reply src/extract_sf.c:248]: Also, I don't understand the reasoning behind the filename change? It used to be only `sf_study_%06d.root`

File: src/extract_sf.c [hunk 17]
```diff
@@ -240,7 +243,14 @@ int run(char *in_filename, bool use_fmt, int nevn, int run_no) {
...
+    } else{
+        sprintf(out_filename, "sf_study_%s", in_filename);
```
Comment (bleaktwig) [inline src/extract_sf.c:250]: This filename is wrong too, it's not saved to the correct directory and it has no extension.

Comment (bleaktwig) [reply src/extract_sf.c:250]: Perhaps instead of adding a lot of if-else statement it would be easier to assign an arbitrary run number to simulations and work from there? It's a bit dirty but as long as it's correctly explained to the user it seems like a much more maintainable solution.

File: src/extract_sf.c [hunk 18]
```diff
@@ -268,7 +278,7 @@ int run(char *in_filename, bool use_fmt, int nevn, int run_no) {
     }
 
     // Write results to file.
-    FILE *t_out = fopen(Form("../data/sf_params_%06d.root", run_no), "w");
+    FILE *t_out = fopen(Form("../root_io/sf_params_%06d.root", run_no), "w");
```
Comment (bleaktwig) [inline src/extract_sf.c:281]: Also this. sampling fraction parameters were saved to `data` instead of `root_io` for a reason. Why is this changed? Additionally, this suffers from the same effect as the sf_study filename since the run_no is invalid.

File: src/hipo2root.c [hunk 19]
```diff
@@ -12,16 +17,22 @@
...
-
-    char *out_filename = (char *) malloc(128 * sizeof(char));
-    sprintf(out_filename, "../root_io/banks_%06d.root", run_no);
-
+    
+    out_filename = (char *) malloc(128 * sizeof(char));
+    if(!use_simul){
+        sprintf(out_filename, "../root_io/banks_%06d.root", run_no);
+    } else{
+        sprintf(out_filename, "%s.root", in_filename);
```
Comment (bleaktwig) [inline src/hipo2root.c:33]: While I agree that my directory convention is awful, its use should be consistent. Please update this to use the `root_io` folder, right now its writing everything to the execution folder.

File: src/io_handler.c (hunk 6 above)
Comment (bleaktwig) [reply src/io_handler.c:68]: nvm, it does run.

File: src/make_ntuples.c [hunk 20]
```diff
@@ -62,14 +62,26 @@ double get_tof(REC_Scintillator rsci, REC_Calorimeter  rcal, int pindex) {
...
+    char*  out_filename = (char *) malloc(128 * sizeof(char));
+    if(!use_simul){
+        // Extract sampling fraction parameters.
+        if (get_sf_params(Form("../data/sf_params_%06d.root", run_no), sf_params)) return 8;
```
Comment (bleaktwig) [inline src/make_ntuples.c:75]: The run number error I mentioned before happens in this file too, causing the program to end its execution early at this point. I cannot execute the program due to this error, so please fix all review comments up to this point so that I can continue the review.

**Review by bleaktwig — CHANGES_REQUESTED** [review]
Please check all pending comments. I cannot execute the code until they are solved so I can't finish the review.

**Comment by emolinac** [comment]
Veré esto en la tardecilla si alcanzo!

Sí, admito que el extract_sf me quedo desordenado porque estaba en un apuro
jeje lo ordenaré y estandarizaré

Chai

File: src/make_ntuples.c (hunk 14 above)
Comment (emolinac) [reply src/make_ntuples.c]: Added this values to process the simulations. Moved the values to constants.C.
Also, added a variation of get_sf_params() that works for simuls and mantains the format of obtaining sf params.

Comment (emolinac) [reply src/make_ntuples.c]: Forgot to mention this: These are sampling fraction parameters obtained from the CCDB.

Comment (bleaktwig) [reply src/make_ntuples.c]: Could you move them to the data folder? You can add an exception to the .gitignore there to allow this. It's a much cleaner alternative and mimics the behaviour used for data.

File: src/extract_sf.c (hunk 15 above)
Comment (emolinac) [reply src/extract_sf.c:230]: Cheked!

File: src/hipo2root.c (hunk 5 above)
Comment (emolinac) [reply src/hipo2root.c]: Fixed!

File: src/io_handler.c (hunk 7 above)
Comment (emolinac) [reply src/io_handler.c]: Fixed! used overload as my weapon of choice.

File: src/make_ntuples.c (hunk 14 above)
Comment (emolinac) [reply src/make_ntuples.c]: you mean to add a piece of code with these values on the data folder?

Comment (bleaktwig) [reply src/make_ntuples.c]: When you run the extract_sf program, you get a simple text file similar to this one:

    0.26300 0.98500 -0.03600 0.00200 0.01660 1.0 0.0 0.0
    0.24022 1.00000 -0.01833 0.00050 0.01825 1.0 0.0 0.0
    0.23976 1.00000 -0.01600 0.00070 0.02249 1.0 0.0 0.0
    0.23774 1.00000 -0.01600 0.00070 0.01816 1.0 0.0 0.0
    0.25118 1.00000 -0.02615 0.00070 0.01735 1.0 0.0 0.0
    0.24044 1.00000 -0.01921 0.00050 0.01817 1.0 0.0 0.0
    0.24210 1.00000 -0.01713 0.00066 0.01646 1.0 0.0 0.0

where each line is a sector, the first four params are sf1 through sf4, and the following params are sfs1 through sfs4. This is the same format you can see on the [CCDB](https://clasweb.jlab.org/cgi-bin/ccdb/show_request?request=/calibration/eb/electron_sf:0:default:2018-06-11_19-11-20).

You can manually write down a file with this same format containing the values from your hardcoded array, and the program will read it no problem.

File: src/extract_sf.c (hunk 16 above)
Comment (emolinac) [reply src/extract_sf.c:248]: Fixed the data problem. In the get_run_no() function I was using use_simul in the ifs instead of *use_simul....
The second point was a taldo. Reverted to the old name convention.

File: src/extract_sf.c (hunk 18 above)
Comment (emolinac) [reply src/extract_sf.c:281]: Reverted to old directory convention.

File: src/extract_sf.c (hunk 17 above)
Comment (emolinac) [reply src/extract_sf.c:250]: Fixed the extension issue.
I like the idea of working with a specific run number for simuls. I have two possible numbers in mind:
1.- 11 : In gemc there are some options taken from certain runs. 11 is one of them. I really doubt that we'll use this program to treat that same run number; therefore, we could use that.
2.- Something weird but identifiable like "101010". I doubt there is a usable run with that same number.

File: src/hipo2root.c (hunk 19 above)
Comment (emolinac) [reply src/hipo2root.c:33]: I do agree on keeping a uniform directory convention for everything; but, at this moment, I'm constantly working with different files of simuls so it is unconvenient to use your directory convention for this purpose :/

File: src/make_ntuples.c (hunk 20 above)
Comment (emolinac) [reply src/make_ntuples.c:70]: Fixed!

File: src/hipo2root.c (hunk 19 above)
Comment (bleaktwig) [reply src/hipo2root.c:33]: How so? In any case, if it's a problem to keep the convention for your personal use then maybe you can keep the repo with the convention while your local version has the change. I can merge the pull request with this change if you really need it but I'll probably end up changing it in a commit anyway out of habit.

File: src/make_ntuples.c (hunk 20 above)
Comment (bleaktwig) [reply src/make_ntuples.c:70]: Great, thanks! I'll be out of town during the weekend but I'll get back to checking the PR on Monday.

File: src/extract_sf.c (hunk 16 above)
Comment (bleaktwig) [reply src/extract_sf.c]: Thanks!

File: src/extract_sf.c (hunk 17 above)
Comment (bleaktwig) [reply src/extract_sf.c]: Check in the software developer documentation, I'm pretty confident there are some run numbers reserved for simulation. If there's no info I can ask around at the lab on Monday.

Comment (emolinac) [reply src/extract_sf.c]: Please ask :D

File: src/hipo2root.c (hunk 19 above)
Comment (emolinac) [reply src/hipo2root.c:33]: To keep the format of the main branch I'll use your format. I'll keep another format for myself in the meantime :)

File: src/extract_sf.c (hunk 17 above)
Comment (emolinac) [reply src/extract_sf.c]: Anyways, I was thinking that for simuls we could use the following format:
6 digits : 999XYZ
Where XYZ is the energy of the simul multiplied by ten. This, in case a simul with fractionary energy is used.
Examples:
Ebeam = 11    Gev  -> Run no : 999110
Ebeam = 10.6 GeV -> Run no : 999106
Ebeam = 22    GeV -> Run no : 999220

Comment (bleaktwig) [reply src/extract_sf.c]: That's not a bad idea. Let's use that standard. If we eventually find out there's another one at the lab we can just switch without much effort.

**Comment by bleaktwig** [comment]
Just tried running again, and I have two issues (before looking at the source code).
* Now the output of `extract_sf` has the .root extension, despite the fact that its a plain text file.
* After running `make_ntuples` I get no output file without any error messages or something of that type. Maybe there's a wrong if around there somewhere?

**Comment by emolinac** [comment]
Ok! I know what is happening with those two points:

- The text file with the parameters of the SF has indeed a .root extension (look at your branch). Actually that was strange for me at first but I let it slide since it works as it is haha. I will change the extension to .txt 
- I just noticed that my make_ntuple has the output file for data with the same name as the file that uses as input (my bad) I'll fix it

File: src/make_ntuples.c [hunk 21]
```diff
@@ -62,22 +62,12 @@ double get_tof(REC_Scintillator rsci, REC_Calorimeter  rcal, int pindex) {
...
-        sprintf(out_filename, "../root_io/banks_%06d.root", run_no);
-    } else{
-        sprintf(out_filename, "ntuple_%s", in_filename);
-    }
+int run(char * in_filename, bool debug, int nevn, int run_no, double beam_E) {
+    double sf_params[NSECTORS][SF_NPARAMS][2];
+    if (get_sf_params(Form("../data/sf_params_%06d.txt", run_no), sf_params)) return 8;
 
+    char*  out_filename = (char *) malloc(128 * sizeof(char));
+    sprintf(out_filename, "../root_io/ntuples_%06d.root", run_no);
```
Comment (bleaktwig) [inline src/make_ntuples.c:70]: This is wrong, but you had no way of knowing. The `ntuples.root` file *intentionally* does not indicate run number, since in the future it will be able to handle various input files with different run numbers. Please change it to how it is in the, master branch: `ntuples.root`.

It would be an useful feature in the future to allow the user to indicate an output filename (and indicate an input filename in `draw_plots`), but its not a priority for now, and I'd rather finish my thesis than worry about this small stuff.

**Review by bleaktwig — CHANGES_REQUESTED** [review]
All looks good except for the comments. Now it runs fine, except for the `ntuples.root` naming convention I mentioned on a comment. Good job!

File: src/make_ntuples.c (hunk 14 above)
Comment (emolinac) [reply src/make_ntuples.c]: done!

File: src/hipo2root.c (hunk 19 above)
Comment (emolinac) [reply src/hipo2root.c]: Substited this system for the simul run system.

File: src/make_ntuples.c (hunk 21 above)
Comment (emolinac) [reply src/make_ntuples.c]: Reverted to original!

File: data/.gitignore [hunk 22]
```diff
@@ -1,2 +1,5 @@
 *
+!sf_params_999106.txt
+!sf_params_999110.txt
+!sf_params_999120.txt
```
[the remark this answers is not in the document]
Comment (emolinac) [reply data/.gitignore:4]: Done! Although I do not know what additional info could be useful apart from the run-number format...

**Comment by emolinac** [comment]
In make_ntuples.C I had to maintain some of the previous structure to be able to process events with and without trigger electrons  and correctly assign leptonic/hadronic variables. At the moment I cannot think of a more elegant/simple way to do it. Any suggestion is welcomed.

Let me know if the make_ntuples output produces the same number of electrons (trigger and not trigger) than a file you already have processed.

**Comment by bleaktwig** [comment]
I see more total particles *and* more trigger electrons in your fork compared to upstream. Maybe you're overcounting? Alternatively, if  you found a bug that makes the upstream lose particles please let me know.

Here you can see more trigger electrons:
Screenshot_20220805_150901

And more particles in general:
Screenshot_20220805_150919

The distributions are fairly similar, but I'm concerned about the difference in the counts.

**Comment by emolinac** [comment]
I understand. I'll dedicate the morning to check what is going on.

**Comment by emolinac** [comment]
Ok, whenever you can test things again and let me know if there are some discrepancies with your nominal files.
- I implemented the algorithm I showed you last meeting.
- Also included the latest updates of your master branch.

**Review by bleaktwig — APPROVED** [review]
Looking good!
